### PR TITLE
Add increase fraction to QS backpressure config

### PR DESCRIPTION
**Pull request by vusirikala**
## Description
Add `increase_fraction` parameter to QS backpressure config. When the QS is no longer in backpressure, we increase the number of transactions pulled from mempool per second by a multiplicative factor.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [x] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [x] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: consensus/src/quorum_store/batch_generator.rs [hunk 1]
```diff
@@ -431,7 +431,7 @@ impl BatchGenerator {
                         if back_pressure_increase_latest.elapsed() >= back_pressure_increase_duration {
                             back_pressure_increase_latest = tick_start;
                             dynamic_pull_txn_per_s = std::cmp::min(
-                                dynamic_pull_txn_per_s + self.config.back_pressure.dynamic_min_txn_per_s,
+                                (dynamic_pull_txn_per_s as f64 * self.config.back_pressure.increase_fraction) as u64,
```
Comment (bchocho) [inline consensus/src/quorum_store/batch_generator.rs:434]: With multiplicative increase + decrease, I'm a bit concerned the limit will thrash around. Can you kick off a forge run that will hit backpressure so we can see how it behaves?

Comment (bchocho) [reply consensus/src/quorum_store/batch_generator.rs:434]: Multiplicative increase + decrease could turn out fine (it's not TCP where you actually lose packets when you over estimate) but I do want to see some experiments and maybe we can tune the factor as well accordingly

Comment (vusirikala) [reply consensus/src/quorum_store/batch_generator.rs:434]: I'm running some experiments on this [PR](https://github.com/aptos-labs/aptos-core/pull/13601) which additionally uses a more intelligent proof queue that tries to count the remaining transactions in proof queue without duplicates.

In case of pipeline backpressure, we add a delay only as long as there is a backpressure. When there is no backpressure, we don't add any delay in creating a proposal.
In case of QS backpressure, with the above mulitplicative increase, we increase the limit back to the original value 2 seconds after the backpressure reduces. I'm hoping we should be fine.

File: config/src/config/quorum_store_config.rs [hunk 2]
```diff
@@ -38,6 +39,7 @@ impl Default for QuorumStoreBackPressureConfig {
             decrease_duration_ms: 1000,
             increase_duration_ms: 1000,
             decrease_fraction: 0.5,
+            increase_fraction: 1.5,
```
Comment (sitalkedia) [inline config/src/config/quorum_store_config.rs:42]: I don't see this being used anywhere - are you missing some changes?

File: mempool/src/core_mempool/mempool.rs [hunk 3]
```diff
@@ -355,6 +355,7 @@ impl Mempool {
                 while skipped.contains(&skipped_txn) {
                     inserted.insert(skipped_txn);
                     result.push(skipped_txn);
+                    skipped.remove(&skipped_txn);
```
Comment (sitalkedia) [inline mempool/src/core_mempool/mempool.rs:358]: Unrelated change?

File: config/src/config/quorum_store_config.rs (hunk 2 above)
Comment (vusirikala) [reply config/src/config/quorum_store_config.rs:42]: Oops. Not sure what happend. I was using the increase_fraction in batch_generator. Pushed the changes.

File: mempool/src/core_mempool/mempool.rs (hunk 3 above)
Comment (vusirikala) [reply mempool/src/core_mempool/mempool.rs:358]: Earlier, we weren't removing `skipped_txn` from skipped even after adding the transaction to `inserted`. This is actually resulting some transactions being added to the batch multiple times. 
As it's just a one line change, I combined it with this PR instead of creating a separate PR.

File: config/src/config/quorum_store_config.rs [hunk 4]
```diff
@@ -32,12 +33,13 @@ impl Default for QuorumStoreBackPressureConfig {
             // validator has recieved Proof of Store but has not yet committed.
             // The batches generated by different validators could have common transactions.
             // Roughly, target TPS * commit latency seconds * duplication factor of a transaction in batches.
-            backlog_txn_limit_count: 36_000,
+            backlog_txn_limit_count: 72_000,
```
Comment (vusirikala) [inline config/src/config/quorum_store_config.rs:36]: This [forge](https://github.com/aptos-labs/aptos-core/actions/runs/9487115246) run at 5k TPS faces a lot of QS backpressure at 5k TPS workload. So, increased the limits.

Comment (sitalkedia) [reply config/src/config/quorum_store_config.rs:36]: We will need to revert this change as this makes Forge runs unstable. See https://github.com/aptos-labs/aptos-core/pull/13666 and https://aptos-org.slack.com/archives/C077LR8EKGA/p1718213547242389?thread_ts=1718124334.521749&cid=C077LR8EKGA

Comment (vusirikala) [reply config/src/config/quorum_store_config.rs]: I reverted back the change. With 12k `backlog_txn_limit_count`, we won't be able to do more than like 2-3k TPS though.

File: consensus/src/quorum_store/batch_generator.rs (hunk 1 above)
Comment (vusirikala) [reply consensus/src/quorum_store/batch_generator.rs:432]: @bchocho I've done the below forge test with the increase fraction and other changes in the PR. The QS backpressure hits here. 

https://github.com/aptos-labs/aptos-core/actions/runs/9559040713
